lightning/src/ln/channel.rs

```diff
@@ -1491,11 +1505,13 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		let local_initial_commitment_tx = self.build_commitment_transaction(self.cur_local_commitment_transaction_number, &local_keys, true, false, self.feerate_per_kw).0;
 		let local_sighash = hash_to_message!(&bip143::SighashComponents::new(&local_initial_commitment_tx).sighash_all(&local_initial_commitment_tx.input[0], &funding_script, self.channel_value_satoshis)[..]);
 
+		let their_funding_pubkey = &self.their_pubkeys.as_ref().unwrap().funding_pubkey;
```

Comment:
If I use the `self.funding_pubkey()` helper here, I get an immutable vs mutable conflict.  Not sure why because there are other calls with immutable self above.

Reply:
Right, the issue is that the helper returns a reference to something in self, so now you have a non-mutable ref for the lifetime of their_funding_pubkey, but then later we try to modify self (taking it as a mutable ref) which ends with us with a mutable and non-mutable ref to self at the same time, which is illegal. I suppose hiding it behind a wrapper changes the semantics to prevent the borrow-individual-member-fields logic.